lib/metrics/reporter.js

```diff
@@ -52,8 +52,10 @@ class MetricsReporter extends Reporter {
         }
       }
 
-      for (const metric of seen.values()) {
-        this._agent._transport.sendMetricSet(metric)
+      if (this._agent._transport) {
```

Comment:
Reviewer note: This was added to not crash in the metrics reporting interval if, in the interim, `agent.destroy()` has been called -- which sets `this._agent._transport` to null.  Similar is already done for `sendSpan` here: https://github.com/elastic/apm-agent-nodejs/blob/ec719c29384ef721d615339e9a6ea3541d15154d/lib/instrumentation/index.js#L268